Makefile

```diff
@@ -128,12 +106,55 @@ run-kube-proxy:
 	-docker rm -f calico-kube-proxy
 	docker run --name calico-kube-proxy -d --net=host --privileged gcr.io/google_containers/hyperkube:v$(K8S_VERSION) /hyperkube proxy --master=http://127.0.0.1:8080 --v=2
 
-kubectl:
-	wget http://storage.googleapis.com/kubernetes-release/release/v$(K8S_VERSION)/bin/linux/amd64/kubectl
-	chmod 755 kubectl
-
 dist/calicoctl:
-	mkdir -p dist
-	sudo chmod a+w dist
-	curl -o dist/calicoctl -L https://github.com/projectcalico/calico-containers/releases/download/v0.17.0/calicoctl
+	curl -o dist/calicoctl -L https://github.com/projectcalico/calico-containers/releases/download/v$(CALICO_NODE_VERSION)/calicoctl
 	chmod +x dist/calicoctl
+
+.PHONY: vendor-up
+vendor-up:
+	glide up -strip-vcs -strip-vendor --update-vendored --all-dependencies
+
+vendor:
+	glide install -strip-vcs -strip-vendor --cache
+
+.PHONY: update-tools
+update-tools:
+	go get -u github.com/Masterminds/glide
+	go get -u github.com/kisielk/errcheck
+	go get -u golang.org/x/tools/cmd/goimports
+	go get -u github.com/golang/lint/golint
+
+.PHONY: static-checks
+static-checks:
+	# Format the code and clean up imports
+	goimports -w *.go utils/*.go ipam/*.go  test_utils/*.go
+
+	# Check for coding mistake and missing error handling
+	go vet -x $(glide nv)
+	errcheck . ./ipam/... ./utils/...
+
+	# Check code style
+	-golint calico.go
+	-golint utils
+	-golint ipam
+
+dist/calico: $(SRCFILES)
+	CGO_ENABLED=0 go build -v --tags "$(BUILD_TAGS)" -o dist/calico \
+	-ldflags "-X main.VERSION=$(shell git describe --tags --dirty)" calico.go;
```

Comment:
The current command produces output like `v1.3.1-5-g9197375-dirty` I think that's enough - it shows the most recent tag and the number of commits since (and the id of the last commit)